src/App.js

```diff
@@ -3,21 +3,41 @@ import './App.css';
 import chatMessages from './data/messages.json';
 import ChatLog from './components/ChatLog';
 
+
 const App = () => {
-  const [likesCount, setLikesCount] = useState(0);
 
-  const increaseLikes = () => {
-    setLikesCount(likesCount + 1);
+  const [messages, setMessages] = useState(chatMessages);
+
+  const handleLikeToggle = (id) => {
+    setMessages(messages => {
+      return messages.map(message => {
+        if (message.id === id) {
+          return {
+            ...message,
+            liked: !message.liked
+          };
+        } else {
+          return message;
+        }
+      });
+    });
   };
 
+  const totalLikeTally = messages.reduce((total, message) => {
+    if (message.liked) {
+      total += 1;
+    };
+    return total;
+  }, 0);
```

Comment:
👍 Nice use of `reduce` to tally the number of liked messages. Personally, I always need to stop and think a litle more carefully whenever I encounter `reduce` in code, but we can help this a bit by hiding the `reduce` call in a helper function.

Making a helper function can really help with our code clarity. Here's an approach of how we might move the `reduce` call to a helper:
```js
const countLikes = (messages) => {
  // adding a bool to a number treats true as 1 and 0 as false
  return messages.reduce((totalLikes, message) => (totalLikes + message.liked), 0);
};
```

Between
```js
  const totalLikeTally = countLikes(messages);
```
and 
```js
  const totalLikeTally = messages.reduce((total, message) => {
    if (message.liked) {
      total += 1;
    };
    return total;
  }, 0);
```
I know which _I_ find more clear, and it's not the one with `reduce`.